engine/src/main/java/org/terasology/world/generation/ScalableFacetProvider.java

```diff
@@ -0,0 +1,12 @@
+// Copyright 2021 The Terasology Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+package org.terasology.world.generation;
+
+public interface ScalableFacetProvider extends FacetProvider {
+    void process(GeneratingRegion region, float scale);
```

Comment:
Can the facet providers assume something about the scale, e.g., the range the value is in? Can they query the min/max scale somehow?

Reply:
Currently the scale is never less than 1, or more than about 1000 (because the LODs slider doesn't go past 10). Do you think that some sort of pre-computed stuff in `setSeed` or `initialize` might be useful? I can't think of any use cases for that information (except that a `ScalableFacetProvider` might rely on the scale never being less than 1, but that's more of an algorithmic simplicity thing, and being able to query it wouldn't really help).

Reply:
I was thinking about a facets that would want to use a simplified model (e.g., just 3 stages of LOD or something like that). However, I think I was missing some pieces about how this works, and that we are not generating placeholder meshes that get scaled to the correct size, but instead try to generate the world (features) in lower resolution. 

I think those are two essentially different ways of doing the LOD effect (your's keeping the blockiness, while the other approach could do more interpolation/low-poly effect). I'm totally fine with what we have here in this PR, just trying to get my head around this. 😉